helix-core/src/main/java/org/apache/helix/cloud/azure/AzureCloudInformationFetcher.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.helix.cloud.azure;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.common.cloud.CloudInformationFetcher;
+import org.apache.helix.common.cloud.SourceType;
+
+public class AzureCloudInformationFetcher implements CloudInformationFetcher <String> {
+
+  public AzureCloudInformationFetcher() {
+  }
+
+  /**
+   * Get the Azure cloud information based on source type
+   * @return Azure cloud information as a string
+   */
+  @Override
+  public String get(SourceType type) {
```

Comment:
How would we get all relevant information based solely on type? Wouldn't we need more metadata?

For example, if the type was a URL - what would be the actual url you should be reading from? What if the cloud information is spread over different types in other types of clouds? Also to support migration efforts, we might have to multiplex multiple cloud systems - will we ever support multiple types in that case (or design this in a flexible enough way for us to enable use cases like that)?

Reply:
I updated the code with constructor. The sources are input of constructor.
We do support multiple types of input in the fetcher. Based on the source type, we have different functions to retrieve the information. All the response will be collected and input to the parser. The reason we do not have 1-1 mapping from input to output is that in that way we will end up with too many types of schemas, and most of the fields are not needed. e.g. in the instance metadata response from Azure, the Json response have more than 30 fields, but we only care about 2 or 3. We will just grab the fields and populate the corresponding fields in Azure computer information file without looking at other fields.